### PR TITLE
Handle has-a relationships for type-hinted arguments in class diagrams

### LINKED ISSUE #4744: pyreverse: has-a relationships for type-hinted arguments
Thanks for your work on this @mbyrnepr2 . 

Quick question: It looks like the use case described in the first post works, but it doesn't show the "has-a" relationship the same way it would for a class instance defined within. 

For example:

Here, where the instance of class B is passed to A's initializer, A's "box" shows it has a member of type B, but doesn't show this graphically.
image

Here, where an instance of B is created in class A's initializer, it does show it graphically.
image

Is there a way have this visual "has a" relationship to work for type-hinted, passed, arguments?

_Originally posted by @BenBlumer in https://github.com/PyCQA/pylint/issues/1548#issuecomment-886099382_

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [ ] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [ ] Write a good description on what the PR does.

## Description
We get a KeyError on this line https://github.com/PyCQA/pylint/blob/main/pylint/pyreverse/diagrams.py#L194 because the value is not of type ClassDef and hence cannot be found in the self._nodes whose keys are all of that type.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Related Issue

Closes #4744

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -14,6 +14,10 @@ What's New in Pylint 2.9.6?
 ===========================
 Release date: TBA
 
+* pyreverse: Show class has-a relationships inferred from the type-hint
```
Comment (Pierre-Sassoulas) [inline ChangeLog:17]: I'd like to put that in 2.10 because it seems like a new feature, what do you think ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Looks nice, could you add one or two automated tests for this, please ?

File: ChangeLog (hunk 1 above)
Comment (mbyrnepr2) [reply ChangeLog:17]: Sounds good to me.

**Comment by mbyrnepr2** [comment]
> Looks nice, could you add one or two automated tests for this, please ?

Agree. Will address asap.

File: ChangeLog (hunk 1 above)
Comment (Pierre-Sassoulas) [reply ChangeLog]: Fixed while fixing the conflict with the main branch :)

**Comment by mbyrnepr2** [comment]
Sorry for the delay @Pierre-Sassoulas @DudeNr33. Added some testing. I will have no access to a laptop for the next couple of days but can respond occasionally if there are any issues.

**Comment by DudeNr33** [comment]
No worries. My PR is nothing urgent, it can wait a bit longer.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for adding the test this looks great !

**Pull request merged**